File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Java/src/NapakalakiGame/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 * @author juanka1995
 */
public class DeathBadConsequence extends NumericBadConsequence{
    
    private boolean death;
    
    public DeathBadConsequence (String text, boolean death){
        super(text,Player.MAXLEVEL,MAXTREASURES,MAXTREASURES);
        this.death = death;
    }
    
    public boolean isDeath() {
        return death;
    }
    
    @Override
    public NumericBadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v, ArrayList<Treasure> h){
        return super.adjustToFitTreasureLists(v,h);
    }
    
    @Override
    public String toString() {
        return super.toString() +  "\ndeath:"  + death;
    }
}

File: AÑO 2/programacion_y_diseño_orientado_a_objetos/Practicas/Napakalaki_Ruby/lib/death_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'numeric_bad_consequence.rb'

class DeathBadConsequence < NumericBadConsequence
  
  attr_accessor :death
  
  def initialize(text)
    super(text,Player.maxLevel,@@MAXTREASURES,@@MAXTREASURES)
    @death = true;
  end
  
  def to_s
    super() + ", death=#{@death}"
  end
end
